lib/connect.serv.js

```diff
@@ -0,0 +1,299 @@
+        }
+      }
+    });
+    return wcProviderNext;
+  }
+}
+
+/**
+ * @notice Checks if a user has an existing connection
+ * @param claimAddress address with GOOD tokens to claim
+ * @return provider instance and connect wallet information
+ */
+
+export async function isConnected(claimAddress) {
+  if (claimAddress !== null) {
+    let providerName, providerInstance, tryMetamask;
+    const web3 = new Web3(Web3.givenProvider || infuraConfig.infuraUrl);
+    const Wc3 = new WalletConnectProvider({
+      infuraId: infuraConfig.infuraId,
+      rpc: {
+        1: infuraConfig.infuraUrl,
+        122: infuraConfig.pocketUrl,
+      }
+    });
+    const web3wc = new Web3(Wc3);
+
+    let tryWalletConnect = Wc3.wc.accounts;
+    if (!tryWalletConnect[0]) {
+      tryMetamask = await web3.eth.getAccounts();
+    }    
+    const isConnected = tryWalletConnect[0] ? 'WC' : tryMetamask[0] ? 'MM' : null;
+    if (isConnected) {
+      providerName = isConnected,
+      providerInstance = tryWalletConnect[0] ? web3wc : web3;
+      const providerInit = walletConnect(providerName, providerInstance, claimAddress);
+      return providerInit;
+    }
+  }
+}
+
+/**
+ * @notice Checks if a user already claimed their tokens on a network
+ * @param currentConnection contains all the necessary information of the connected wallet
+ * @return boolean's for claim-status per network
+ */
+export async function getClaimStatus(currentConnection) {
+  let contractAddressess = {
+        fuse: '0x3A9299BE789ac3730e4E4c49d6d2Ad1b8BC34DFf',
+        rootState: '0x3A9299BE789ac3730e4E4c49d6d2Ad1b8BC34DFf'
+      },
+      claimStatus = {
+        fuse: false,
+        rootState: false
+      };
+
+  for (const [network, address] of Object.entries(contractAddressess)){
+    let web3 = new Web3( (network == 'rootState' ? infuraConfig.pocketUrl : infuraConfig.infuraUrl));
+    const latest = await web3.eth.getBlockNumber();
+    const gRepContract = new web3.eth.Contract(GReputationABI.abi, address);
+    let idHash = web3.utils.keccak256(network);
+    let getVotes = await gRepContract.methods.getVotesAtBlockchain(idHash, currentConnection.connectedAddress, latest).call();
+    if (getVotes > 0){
+      claimStatus[network] = true;
+    }
+  }
+
+  return claimStatus;
+}
+
+
+/**
+ * @notice gets the current set recipient for _user from contract.
+ * @param contractInstance contract instance for connected network. is set when a user is clicking through tabs,
+ * is null on first mount.
+ * @param currentConnection necessary information of the connected wallet
+ * @return contractInstance and current recipient. Is empty address when none is set (default: _user)
+ */
+export async function getRecipient(contractInstance, currentConnection) {
+  const web3 = new Web3(new Web3.providers.HttpProvider(
+      currentConnection.chainId == 122 ? infuraConfig.pocketUrl : infuraConfig.infuraUrl)),
+      contractAddr = '0x3A9299BE789ac3730e4E4c49d6d2Ad1b8BC34DFf';
```

Comment:
shouldnt be hardcoded
anything else is hardcoded?

Reply:
done